src/list.ts

```diff
@@ -0,0 +1,259 @@
+    });
+
+    const productName = document.createElement('input');
+    productName.type = 'text';
+    productName.disabled = true;
+    productName.value = data.artikelname;
+
+    const quantity = document.createElement('input');
+    quantity.type = 'number';
+    quantity.min = '1';
+    quantity.max = '999';
+    quantity.value = String(data.anzahl);
+    quantity.disabled = true;
+
+    const editButton = document.createElement('button');
+    editButton.innerHTML = '&#9998';
+
+    const deleteButton = document.createElement('button');
+    deleteButton.innerHTML = '&#x1F5D1;';
+
+    deleteButton.addEventListener('click', async (event) => {
+        event.preventDefault();
+
+        database.deleteDocument(
+            CONFIG.DATABASE_ID,
+            CONFIG.DB_COLLECTION_SHOPLISTENTRY,
+            data.$id!
+        ).then(() => { },
+            () => {
+                window.alert('Das ausgewählte Produkt konnte nicht gelöscht werden. :(');
+            });
+    });
+
+    editButton.addEventListener('click', async (event) => {
+        event.preventDefault();
+        if (!productName.disabled.valueOf()) {
+            data.anzahl = Number.parseInt(quantity.value);
+            data.artikelname = productName.value;
+            await updateListEntry(data);
+        }
+        productName.disabled = !productName.disabled.valueOf();
+        quantity.disabled = !quantity.disabled.valueOf();
+    });
+
+    addToRow(checkbox, row);
+    addToRow(productName, row);
+    addToRow(quantity, row);
+    addToRow(editButton, row);
+    addToRow(deleteButton, row);
+
+    return row;
+}
+
+function addToRow(element: HTMLElement, row: HTMLTableRowElement) {
+    const td = document.createElement('td');
+    td.appendChild(element);
+    row.appendChild(td);
+}
+
+function addRowToTable(newEntry: Listeneintrag, id?: string) {
+    if (id == undefined) {
+        productTable.appendChild(createTableRow(newEntry));
+        return;
+    }
+    productTable.appendChild(createTableRow(newEntry, id));
+}
+
+function realtimeUpdate(list: Einkaufsliste) {
+    const listChannel = 'databases.' + CONFIG.DATABASE_ID +
+        '.collections.' + CONFIG.DB_COLLECTION_SHOPPINGLISTS +
+        '.documents.' + list.$id!;
+    const documentChannel = 'databases.' + CONFIG.DATABASE_ID +
+        '.collections.' + CONFIG.DB_COLLECTION_SHOPLISTENTRY +
+        '.documents';
+    client.subscribe([listChannel, documentChannel], response => {
+        if (response.channels.includes(listChannel)) {
+            const changedList = response.payload as Einkaufsliste;
+            updateListInfo(changedList);
+        }
+        if (response.channels.includes(documentChannel)) {
+            documentEventHandler(response, list.$id!);
+        }
+    });
+}
+
+function documentEventHandler(response: RealtimeResponseEvent<unknown>, id: string) {
+    const changedEntry = response.payload as Listeneintrag;
+    if (changedEntry.$id != id) {
```

Comment:
You're comparing the id of the shopping list against the id of a shopping list entry. I don't think that makes sense.

As far as I can tell, the ids never match, which means the function always returns.